### DISCUSSION TITLE
Projector and Reactor discovery fails due to incorrect namespace resolution

### DISCUSSION THREAD

**Discussion by AzaZPPL** [spatie/laravel-event-sourcing#456] [Ideas]
Hello,

I'm currently integrating this package into a project where the application's root folder is configured differently from the Laravel default. Specifically, my application's root namespace is `Efekan`, and the source code resides in `/src/` instead of the usual `/app/` directory. I have configured the Composer autoloader accordingly to reflect this structure:

```json
"autoload": {
    "psr-4": {
        "App\\": "app/",
        "Database\\Factories\\": "database/factories/",
        "Database\\Seeders\\": "database/seeders/",
        "Efekan\\": "src/"
    }
}
```

Additionally, my `event-sourcing` config is set to auto-discover projectors and reactors within the `/src/User/Domain/Projector` directory:

```php
'auto_discover_projectors_and_reactors' => [
    base_path('/src/User/Domain/Projector'),
],
'auto_discover_base_path' => base_path() . '/src',
```

Despite this setup, the auto-discovery mechanism fails to locate a projector class located within the specified directory. Debugging led me to `\Spatie\EventSourcing\Support\DiscoverEventHandlers`, where I noticed the discovery process does not correctly pick up the namespace of the file.

I propose introducing a new configuration key, `auto_discover_root_namespace`, to specify the root namespace explicitly for projects that do not use the default `App\` namespace. This could be integrated within the `EventSourcingServiceProvider` as follows:

```php
(new DiscoverEventHandlers())
    ->within(config('event-sourcing.auto_discover_projectors_and_reactors'))
    ->useBasePath(config('event-sourcing.auto_discover_base_path', base_path()))
    ->ignoringFiles(Composer::getAutoloadedFiles(base_path('composer.json')))
    ->useRootNamespace(config('event-sourcing.auto_discover_root_namespace', app()->getNamespace()))
    ->addToProjectionist($projectionist);
```

This adjustment has resolved the issue in my local environment, enabling the correct discovery of event handlers.

**Comment by sebastiandedeyne** [comment]
I'd accept a PR that adds this. Although not even sure it should be configurable, maybe we should just use `app()->getNamespace()` as the default root namespace?
